src/algorithms/all_neighbors.c

```diff
@@ -17,12 +17,15 @@ static void _AllNeighborsCtx_CollectNeighbors
 	ctx->current_level++;
 	GxB_MatrixTupleIter *iter;
 	if(ctx->current_level == array_len(ctx->levels)) {
-		GxB_MatrixTupleIter_new(&iter, ctx->M);
-		array_append(ctx->levels, iter);
-	} else {
-		iter = ctx->levels[ctx->current_level];
+		ASSERT(ctx->M);
+		ctx->levels = array_grow(ctx->levels, 1);
+		// make sure matrix is not bitmap or full
+		GxB_set(ctx->M, GxB_SPARSITY_CONTROL, GxB_SPARSE) ;
+		GxB_MatrixTupleIter_reuse(&ctx->levels[ctx->current_level], ctx->M);
```

Comment:
We usually use `array_append` in these type of situations
I feel like this re-write is a bit cleaner/clearer
```
GxB_MatrixTupleIter iter;
GxB_MatrixTupleIter_reuse(&iter, ctx->M);
array_append(ctx->levels, iter);
```